car_information/car_information_service.py

```diff
@@ -20,11 +20,39 @@ def identify_license_plate(path, image):
     try:
         r = requests.post(OPEN_ALPR_ENDPOINT, data=img_base64)
         result_json = r.json()
+        results = result_json["results"]
 
-        return result_json["results"][0]["plate"]
-    except:
-        return None
+        if len(results) == 0:
+            raise LicensePlateNotDetectedException("Could not detect any license plates")
 
+        license_plate = get_largest_plate(results)
+
+        if not is_valid_license_plate(license_plate):
+            raise IllegalLicensePlateException("The license plate does not follow Norwegian convention")
+
+        return license_plate
+    finally:
+        pass
+
+
+def get_largest_plate(result_json):  # If several plates are in the picture, choose the on that takes the most space
```

Comment:
Maybe move this comment above line 28 instead, as it seems more natural to explain the reasoning for why the function is used/called in that position.

Reply:
Agreed. Done